File: ExemploAutoCompleteTextView/app/src/main/java/nuvemapp/com/br/exemploautocompletetextview/domain/State.java
package nuvemapp.com.br.exemploautocompletetextview.domain;

import java.util.HashMap;

import nuvemapp.com.br.exemploautocompletetextview.R;


public class State {
	private String name;
	private int imgResource;
	public static HashMap<String, Integer> imgMap = new HashMap<String, Integer>();
	static {
        imgMap.put("Acre (AC)", R.drawable.acre);
        imgMap.put("Alagoas (AL)", R.drawable.alagoas);
        imgMap.put("Amapá (AP)", R.drawable.amapa);
        imgMap.put("Amazonas (AM)", R.drawable.amazonas);
        imgMap.put("Bahia (BA)", R.drawable.bahia);
        imgMap.put("Ceará (CE)", R.drawable.ceara);
        imgMap.put("Distrito Federal (DF)", R.drawable.distrito_federal);
        imgMap.put("Espírito Santo (ES)", R.drawable.espirito_santo);
        imgMap.put("Goiás (GO)", R.drawable.goias);
        imgMap.put("Maranhão (MA)", R.drawable.maranhao);
        imgMap.put("Mato Grosso (MT)", R.drawable.mato_grosso);
        imgMap.put("Mato Grosso do Sul (MS)", R.drawable.mato_grosso_do_sul);
        imgMap.put("Minas Gerais (MG)", R.drawable.minas_gerais);
        imgMap.put("Pará (PA)", R.drawable.para);
        imgMap.put("Paraíba (PB)", R.drawable.paraiba);
        imgMap.put("Paraná (PR)", R.drawable.parana);
        imgMap.put("Pernambuco (PE)", R.drawable.pernambuco);
        imgMap.put("Piauí (PI)", R.drawable.piaui);
        imgMap.put("Rio de Janeiro (RJ)", R.drawable.rio_de_janeiro);
        imgMap.put("Rio Grande do Norte (RN)", R.drawable.rio_grande_do_norte);
        imgMap.put("Rio Grande do Sul (RS)", R.drawable.rio_grande_do_sul);
        imgMap.put("Rondônia (RO)", R.drawable.rondonia);
        imgMap.put("Roraima (RR)", R.drawable.roraima);
        imgMap.put("Santa Catarina (SC)", R.drawable.santa_catarina);
        imgMap.put("São Paulo (SP)", R.drawable.sao_paulo);
        imgMap.put("Sergipe (SE)", R.drawable.sergipe);
        imgMap.put("Tocantins (TO)", R.drawable.tocantins);
        imgMap.put("Ontario (ON)", R.drawable.ontario);
        imgMap.put("Quebec (QC)", R.drawable.quebec);
        imgMap.put("Nova Scotia (NS)", R.drawable.nova_scotia);
        imgMap.put("New Brunswick (NB)", R.drawable.new_brunswick);
        imgMap.put("Manitoba (MB)", R.drawable.manitoba);
        imgMap.put("British Columbia (BC)", R.drawable.british_columbia);
        imgMap.put("Prince Edward Island (PE)", R.drawable.prince_edward_island);
        imgMap.put("Saskatchewan (SK)", R.drawable.saskatchewan);
        imgMap.put("Alberta (AB)", R.drawable.alberta);
        imgMap.put("Newfoundland and Labrador (NL)", R.drawable.newfoundland_and_labrador);
    }
	
	
	
	public State(){}
	public State(String name, int imgResource){
		this.name = name;
		this.imgResource = imgResource;
	}
	
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public int getImgResource() {
		return imgResource;
	}
	public void setImgResource(int imgResource) {
		this.imgResource = imgResource;
	}
	
	
	public String toString() {
		return(this.getName());
	}
}

File: ExemploAutoCompleteTextView/PHP/state.php
<?php
	$array = array();
	$text = isset($_POST['search']) ? $_POST['search'] : $_GET['search'];
	$country = isset($_POST['country']) ? $_POST['country'] : $_GET['country'];
	//sleep(1);
	
	if($country == 1){
		$array = array(array("state"=>"Acre (AC)"),
			array("state"=>"Alagoas (AL)"),
			array("state"=>"Amapá (AP)"),
			array("state"=>"Amazonas (AM)"),
			array("state"=>"Bahia (BA)"),
			array("state"=>"Ceará (CE)"),
			array("state"=>"Distrito Federal (DF)"),
			array("state"=>"Espírito Santo (ES)"),
			array("state"=>"Goiás (GO)"),
			array("state"=>"Maranhão (MA)"),
			array("state"=>"Mato Grosso (MT)"),
			array("state"=>"Mato Grosso do Sul (MS)"),
			array("state"=>"Minas Gerais (MG)"),
			array("state"=>"Pará (PA)"),
			array("state"=>"Paraíba (PB)"),
			array("state"=>"Paraná (PR)"),
			array("state"=>"Pernambuco (PE)"),
			array("state"=>"Piauí (PI)"),
			array("state"=>"Rio de Janeiro (RJ)"),
			array("state"=>"Rio Grande do Norte (RN)"),
			array("state"=>"Rio Grande do Sul (RS)"),
			array("state"=>"Rondônia (RO)"),
			array("state"=>"Roraima (RR)"),
			array("state"=>"Santa Catarina (SC)"),
			array("state"=>"São Paulo (SP)"),
			array("state"=>"Sergipe (SE)"),
			array("state"=>"Tocantins (TO)"));
	}
	else{
		$array = array(array("state"=>"Ontario (ON)"),
			array("state"=>"Quebec (QC)"),
			array("state"=>"Nova Scotia (NS)"),
			array("state"=>"New Brunswick (NB)"),
			array("state"=>"Manitoba (MB)"),
			array("state"=>"British Columbia (BC)"),
			array("state"=>"Prince Edward Island (PE)"),
			array("state"=>"Saskatchewan (SK)"),
			array("state"=>"Alberta (AB)"),
			array("state"=>"Newfoundland and Labrador (NL)"));
	}
	
	
	$qtdText = strlen($text);
	$arrayAux = array();
	
	for($i = 0, $tam = count($array); $i < $tam; $i++){
		$textWithoutSignal = preg_replace('/[áàäâãª]/', 'a', substr($array[$i]['state'], 0, $qtdText));
		$textWithoutSignal = preg_replace('/[íìïî]/', 'i', $textWithoutSignal);
		$textWithoutSignal = preg_replace('/[éèêë]/', 'e', $textWithoutSignal);
		$textWithoutSignal = preg_replace('/[óòöôõº]/', 'o', $textWithoutSignal);
		$textWithoutSignal = preg_replace('/[úùüû]/', 'u', $textWithoutSignal);
		$textWithoutSignal = preg_replace('/[ç]/', 'c', $textWithoutSignal);
		$textWithoutSignal = preg_replace('/[ñ]/', 'n', $textWithoutSignal);
	
		if(strcasecmp(substr($array[$i]['state'], 0, $qtdText), $text) == 0
			|| strcasecmp($textWithoutSignal, $text) == 0){
			$arrayAux[] = $array[$i];
		}
	}
	
	echo json_encode($arrayAux);
?>